generate-release/src/migration_guides.rs

```diff
@@ -112,45 +97,62 @@ pub fn generate_migration_guides(
             // 64 is completely arbitrary but felt long enough and is a nice power of 2
             file_name.truncate(64);
 
-            // Generate the metadata block for this migration
-            // We always re-generate it because we need to keep the ordering if a new migration is added
-            let metadata_block = generate_metadata_block(&title, &file_name, &area, pr.number);
+            // Add the markdown extension
+            file_name = format!("{file_name}.md");
 
-            let file_path = path.join(format!("{file_name}.md"));
+            let file_path = path.join(&file_name);
 
             if write_migration_file(
                 &file_path,
                 pr.body.as_ref().context("PR has no body")?,
                 pr.number,
             )? {
-                guides_metadata.push(metadata_block);
+                guides_metadata.push(MigrationGuide {
+                    title,
+                    prs: vec![pr.number],
+                    areas: area.clone().into_iter().collect(),
+                    file_name,
+                });
             }
         }
     }
 
-    if !new_prs {
-        return Ok(());
-    }
+    // Sort by: Area ASC (empty areas at the end), Title ASC
+    guides_metadata.sort_by(|a, b| {
+        let areas_cmp = match (a.areas.is_empty(), b.areas.is_empty()) {
+            (false, false) => {
+                let a_areas = a.areas.clone().into_iter().collect::<Vec<_>>().join(" ");
+                let b_areas = b.areas.clone().into_iter().collect::<Vec<_>>().join(" ");
+
+                a_areas.cmp(&b_areas)
+            }
+            (false, true) => std::cmp::Ordering::Less,
+            (true, false) => std::cmp::Ordering::Greater,
+            (true, true) => std::cmp::Ordering::Equal,
+        };
+
+        areas_cmp.then_with(|| a.title.cmp(&b.title))
+    });
+
+    // Replace and overwrite file.
```

Comment:
```suggestion
    // Create the metadata file, and overwrite it if it already exists.
    //
    // Note:
    // The file, while overwritten,
    // may still contain the same underlying data gotten from
    // the preexisting metadata earlier, if overwrite_existing is false,
    // thus preserving the data even if the file itself is overwritten.
```

Just wanted to add a clarifying comment, so that no one thinks that this is also overwriting the data (unless overwrite_existing was true earlier), not just the file itself.